lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -1607,6 +1607,8 @@ balanceTransaction
     => Tracer m WalletLog
     -> TransactionLayer k ktype SealedTx
     -> ArgGenChange s
+    -> Maybe ([(TxIn, TxOut)] -> Map TxIn (CA.Script KeyHash))
+    -> Maybe ScriptTemplate
```

Comment:
I'm not sure how these should look ultimately: maybe we just need a `ScriptHash -> Script` lookup function + having `balanceTransaction` inspect the addresses of the inputs. Or maybe we want to pass in some `CoinSelection` record / function which can encapsulate the required differences between selecting from shared and non-shared wallets. Regardless, as said, happy to be going with this for the moment 👍